### ISSUE TITLE
[Bug]: `setContentProtection` affects an application window behavior

### ISSUE THREAD

**Issue by KazuyaHayashi** [electron/electron#31735]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

15.2.0

### What operating system are you using?

Windows

### Operating System Version

Windows 10 version 20H2

### What arch are you using?

x64

### Last Known Working Electron version

15.1.2

### Expected Behavior

Apply `setContentProtection` function without any effects on the application window behavior on `browser-window-created` event.

### Actual Behavior

When `setContentProtection` function is called on `browser-window-created` event, the application window form will changed when the focus is backed.  The value of the argument of `setContentProtection` has no effect on the result.
![setContentProtectionIssue](https://user-images.githubusercontent.com/441950/140640295-ca25b82d-d36b-4101-b38e-b8def46bab3e.gif)

### Testcase Gist URL

https://gist.github.com/KazuyaHayashi/33372ae657dde602e68fdb50aabafa31

### Additional Information

When I call `setContentProtection` just after creating `BrowserWindow`, this issue does not happen. But I want to set contents protection on each windows which are created by `new-window` event. Therefor, I have called `setContentProtection` function on `browser-window-created` event.

**Comment by deermichel** [comment]
Reproduced on Win10 2004, introduced by https://github.com/electron/electron/pull/31340. Doesn't occur when using `setOpacity` though 🤔

**Comment by deermichel** [comment]
Actually serves as a temporary workaround:
```js
app.on('browser-window-created', function(e, w) {
  w.setOpacity(1.0)
  w.setContentProtection(true)
})
```

**Comment by deermichel** [comment]
Fixed and confirmed to work in 16.0.0.
